generators/lib/distros/octane.js

```diff
@@ -31,21 +32,29 @@ function init() {
     done();
   };
 
-  module.modifyComposer = function(yo, options, localComposer, isNewProject, done) {
+  module.loadComposer = function(yo, options) {
+    var composer = {};
+    var file = yo.templatePath('drupal/drupal/' + options.drupalDistroRelease + '/composer.json');
+    if (gadget.fsExistsSync(file)) {
+      composer = yo.fs.readJSON(file);
+      // Octane gets it's version of Drupal Core from Lightning.
+      // To minimize conflicts with the rest of the generator process, we defer
+      // removing this until the last time Octane can reasonably impose this logic.
+      delete composer.require['drupal/core'];
```

Comment:
The comment is from when I did have it in modifyComposer, so I should probably move this back or change the comment. I pulled it here into the load functionality because I found it was much easier to break the assumptions of Octane not having a drupal/core itself in the main index.js while this was present, since there's processing that happens if it is set.